src/forms/helpers/dropdown-select.js

```diff
@@ -46,8 +57,9 @@ DropdownSelect.propTypes = propTypes
 
 DropdownSelect.defaultProps = defaultProps
 
-function getLabel(values) {
-  return values.length ? values.join(', ') : 'None'
+function getLabel(values, options) {
+  const labels = values.map((v) => options.find((o) => o.value === v).key)
+  return values.length ? labels.join(', ') : 'None'
```

Comment:
While `labels.length` should always equal `values.length`, it's safer to have the guard on `labels.length`